### ISSUE TITLE
Dlib support

### ISSUE THREAD

**Issue by tulliopizzuti** [chaquo/chaquopy#46]
Hi, i'm trying to import dlib.py, dlib.pyc and dlib.so but i get this error when import dlib:
```File "stdlib/threading.py", line 916, in _bootstrap_inner
  File "stdlib/threading.py", line 864, in run
  File "/android_asset/chaquopy/app.zip/chaquopy/utils/console.py", line 14, in <lambda>
    threading.Thread(target=lambda: runnable.run()).start()
  File "class.pxi", line 696, in java.chaquopy.JavaMethod.__get__.lambda2
  File "class.pxi", line 716, in java.chaquopy.JavaMethod.__call__
  File "class.pxi", line 806, in java.chaquopy.JavaMethod.call_nonvirtual_method
  File "env.pxi", line 242, in java.chaquopy.CQPEnv.CallNonvirtualVoidMethodA
  File "env.pxi", line 484, in java.chaquopy.CQPEnv.check_exception
com.chaquo.python.PyException: ImportError: dlopen failed: library "/android_asset/chaquopy/app.zip/dlib.so" not found
at <python>.load_dynamic(stdlib/imp.py:343)
at <python>.__bootstrap__(/android_asset/chaquopy/app.zip/dlib.py:6)
at <python>.<module>(/android_asset/chaquopy/app.zip/dlib.py:7)
at <python>.load_module_impl(/data/user/0/com.chaquo.python.console/files/chaquopy/bootstrap.zip/java/android/importer.py:348)
at <python>.load_module(/data/user/0/com.chaquo.python.console/files/chaquopy/bootstrap.zip/java/android/importer.py:270)
at <python>.java.chaquopy.import_override(import.pxi:19)
at <python>.reraise(/data/user/0/com.chaquo.python.console/files/chaquopy/bootstrap.zip/java/_vendor/six.py:686)
at <python>.java.chaquopy.import_override(import.pxi:60)
at <python>.<module>(/android_asset/chaquopy/app.zip/main.py:6)
at <python>.load_module_impl(/data/user/0/com.chaquo.python.console/files/chaquopy/bootstrap.zip/java/android/importer.py:348)
at <python>.load_module(/data/user/0/com.chaquo.python.console/files/chaquopy/bootstrap.zip/java/android/importer.py:270)
at <python>._load_backward_compatible(<frozen importlib._bootstrap>:626)
at <python>._load_unlocked(<frozen importlib._bootstrap>:656)
at <python>._find_and_load_unlocked(<frozen importlib._bootstrap>:955)
at <python>._find_and_load(<frozen importlib._bootstrap>:971)
at <python>._gcd_import(<frozen importlib._bootstrap>:994)
at <python>.import_module(stdlib/importlib/__init__.py:126)
at <python>.chaquopy_java.Java_com_chaquo_python_Python_getModule(chaquopy_java.pyx:145)
at com.chaquo.python.Python.getModule(Native Method)
at com.chaquo.python.console.MainActivity$Task.run(MainActivity.java:20)
```

How can i access in the /android_asset.... directory? 
If i compile dlib with ndk and copy the *.so file in this directory it will work?

**Comment by mhsmith** [comment]
Duplicate of #33.

If you've compiled dlib.so with the NDK, then you may be able to get it to work this way. However, if you can share your build process with us, then we'd prefer to adopt it as an official Chaquopy package, which would be much easier for everyone.

On the other hand, if you've compiled dlib.so as a regular Linux build, then it's unlikely to work on Android.

**Comment by tulliopizzuti** [comment]
If I succeed I will share the process with pleasure, we share the code for this too! But in what directory should I place the .so file?

**Comment by mhsmith** [comment]
Put it directly in src/main/python, and rename your dlib.py to something else. Then you should just be able to do `import dlib`.

**Comment by tulliopizzuti** [comment]
```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "stdlib/threading.py", line 916, in _bootstrap_inner
  File "stdlib/threading.py", line 864, in run
  File "/android_asset/chaquopy/app.zip/chaquopy/utils/console.py", line 14, in <lambda>
    threading.Thread(target=lambda: runnable.run()).start()
  File "class.pxi", line 696, in java.chaquopy.JavaMethod.__get__.lambda2
  File "class.pxi", line 716, in java.chaquopy.JavaMethod.__call__
  File "class.pxi", line 806, in java.chaquopy.JavaMethod.call_nonvirtual_method
  File "env.pxi", line 242, in java.chaquopy.CQPEnv.CallNonvirtualVoidMethodA
  File "env.pxi", line 484, in java.chaquopy.CQPEnv.check_exception
com.chaquo.python.PyException: ImportError: dynamic module does not define module export function (PyInit_dlib)
at <python>.load_dynamic(stdlib/imp.py:343)
at <python>.load_module_impl(/data/user/0/com.chaquo.python.console/files/chaquopy/bootstrap.zip/java/android/importer.py:409)
at <python>.load_module(/data/user/0/com.chaquo.python.console/files/chaquopy/bootstrap.zip/java/android/importer.py:270)
at <python>.java.chaquopy.import_override(import.pxi:19)
at <python>.reraise(/data/user/0/com.chaquo.python.console/files/chaquopy/bootstrap.zip/java/_vendor/six.py:686)
at <python>.java.chaquopy.import_override(import.pxi:60)
at <python>.<module>(/android_asset/chaquopy/app.zip/main.py:6)
at <python>.load_module_impl(/data/user/0/com.chaquo.python.console/files/chaquopy/bootstrap.zip/java/android/importer.py:348)
at <python>.load_module(/data/user/0/com.chaquo.python.console/files/chaquopy/bootstrap.zip/java/android/importer.py:270)
at <python>._load_backward_compatible(<frozen importlib._bootstrap>:626)
at <python>._load_unlocked(<frozen importlib._bootstrap>:656)
at <python>._find_and_load_unlocked(<frozen importlib._bootstrap>:955)
at <python>._find_and_load(<frozen importlib._bootstrap>:971)
at <python>._gcd_import(<frozen importlib._bootstrap>:994)
at <python>.import_module(stdlib/importlib/__init__.py:126)
at <python>.chaquopy_java.Java_com_chaquo_python_Python_getModule(chaquopy_java.pyx:145)
at com.chaquo.python.Python.getModule(Native Method)
at com.chaquo.python.console.MainActivity$Task.run(MainActivity.java:20)
at com.chaquo.python.utils.ConsoleActivity$Task$1.run(ConsoleActivity.java:367)
```

**Comment by tulliopizzuti** [comment]
Same error using 2.7.15:
ImportError: dynamic module does not define init function (initdlib)

**Comment by mhsmith** [comment]
Please explain how you built the file, and either email the file to contact@chaquo.com, or upload it somewhere and post the link.

**Comment by mhsmith** [comment]
The file was the bare dlib library, not the Python module. If you work out a process for building the Python module, please add details to #33.

**Comment by mhsmith** [comment]
dlib is now available: see #33.
